### PR TITLE
feat(clients) Add updateEndpoint API

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

This PR adds the Pinpoint client `updateEndpoint` API

#### Description of how you validated changes
Locally replaced API calls from Analytics and verified network request performed as expected

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jimblanc — APPROVED** [review]
Looks good to me!

File: packages/core/src/AwsClients/Pinpoint/base.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * The service name used to sign requests if the API requires authentication.
+ */
+const SERVICE_NAME = 'mobiletargeting';
+
+/**
+ * The endpoint resolver function that returns the endpoint URL for a given region.
+ */
+const endpointResolver = (endpointOptions: { region: string }) => ({
+	url: new URL(`https://pinpoint.${endpointOptions.region}.amazonaws.com`),
```
Comment (AllanZhengYP) [inline packages/core/src/AwsClients/Pinpoint/base.ts:21]: Can you confirm if we need to handle CN regions?

File: packages/core/src/clients/handlers/signed.ts
Comment (AllanZhengYP) [inline packages/core/src/clients/handlers/signed.ts:1]: I'd prefer name the handler as `authenticated.ts`

File: packages/core/src/AwsClients/Pinpoint/base.ts (hunk 1 above)
Comment (cshfang) [reply packages/core/src/AwsClients/Pinpoint/base.ts:21]: I don't think Pinpoint is offered in CN regions. But we should be integrating the DNS suffix util either way right?

**Review by AllanZhengYP — APPROVED** [review]
💯 Just don't forget to rename the title to conventional before merging the PR.

**Comment by bergmorten** [comment]
Why have not this been merged into V6 only V5?

[
updateEndpoint in V5](https://docs.amplify.aws/javascript/prev/build-a-backend/more-features/analytics/update-endpoint/)

**Pull request merged**